File: Java/Arrays/Code12.java
// Binary Search
import java.util.Scanner;

public class Code12 {
    public static int binarysearch(int numbers[], int key) {
        int start = 0;
        int end = numbers.length - 1;
        int mid;
        
        while (start <= end) {
            mid = (start + end) / 2;
            
            if (numbers[mid] == key) {
                return mid;  // Found the key, return the index
            }
            else if (numbers[mid] > key) {
                end = mid - 1;  // Adjust the end to search the left half
            }
            else {
                start = mid + 1;  // Adjust the start to search the right half
            }
        }
        return -1;  // If the key is not found
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int numbers[] = {2, 4, 6, 8, 10, 12, 14};
        int key = 6;

        int index = binarysearch(numbers, key);
        if (index != -1) {
            System.out.println("The index at which key is = " + index);
        } else {
            System.out.println("Key not found");
        }

        sc.close();
    }
}

File: Python/Basics2/Code12.py
#Binary Search

def binarySearch(n,tar):
    start = 0
    end = len(n) - 1
    while(start <= end):
        mid = start + (end - start)// 2
        if(tar == n[mid]):
            print(f"target found at index {mid}")
            break
        elif(tar > n[mid]):
            start = mid + 1
        else:
            end = mid - 1
    
n = [99,56,78,43,27]
n.sort()
tar = 78
binarySearch(n,tar)